homeassistant/components/somfy/__init__.py

```diff
@@ -92,6 +93,10 @@ async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
     async def _update_all_devices():
         """Update all the devices."""
         devices = await hass.async_add_executor_job(data[API].get_devices)
+        previous_devices = data[COORDINATOR].data
```

Comment:
Is there no better way than checking the "internals" of the update coordinator here? There is a risk that a behavior change of the update coordinator will break this. Doubly since there is no test.

My point is that this function doesn't control the setting of .data of the coordinator directly. I think I'd have preferred a separate variable holding the success of previous request.

Reply:
I understand your doubt. The data property is part of the contract offer by the DataUpdateCoordinator: https://developers.home-assistant.io/docs/integration_fetching_data 
So, it cannot be changed without any announcement. Create a subclass of the DataUpdateCoordinator would be overkill in this case IMO.

Reply:
Well you could have stored a attribute on the callback function. Or used a captured nonlocal variable for the state.
But I won't require any change. Just a small concern.

Reply:
Thx for merging this PR.

>Well you could have stored a attribute on the callback function. 

Can you please detailed?

Reply:
https://www.kite.com/python/answers/how-to-declare-a-static-variable-in-a-function-in-python